pilot/pkg/config/kube/gateway/deploymentcontroller_test.go

```diff
@@ -72,6 +73,38 @@ func TestConfigureIstioGateway(t *testing.T) {
 			ControllerName: k8s.GatewayController(features.ManagedGatewayController),
 		},
 	}
+	upgradeDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-upgrade",
+			Namespace: "default",
+			Labels: map[string]string{
+				"gateway.istio.io/managed": "istio.io-mesh-controller",
```

Comment:
This needs the `istio.io/gateway-name` label to accurately test the change

Reply:
done